python/location/lbs/activity/audioset/dataprocessing/audio_processing.py

```diff
@@ -0,0 +1,262 @@
+  # extract zero_crossing_rate
+  f_dict["zero_crossing_rate"] = librosa.feature.zero_crossing_rate(y, sr)
+  feature_dict[video_id] = f_dict
+  print("extracted features")
+
+
+def create_csv_from_list():
+  # Creates a csv file to input into TensorFlow with the following structure: is_gunshot, video_id, feature list
+  # input: None
+  # output: None
+  with open("tfdataset.csv", 'w', newline='') as csvfile:
+    csv_writer = csv.writer(csvfile, delimiter=':', quoting=csv.QUOTE_MINIMAL)
+    csv_writer.writerow(
+      ['label', 'video_id', 'chroma_stft', 'chroma_cqt', 'chroma_cens', 'melspectrogram', 'mfcc', 'rms',
+       'spectral_centroid', 'spectral_bandwidth', 'spectral_contrast', 'spectral_flatness', 'spectral_rolloff',
+       'poly_features', 'tonnetz', 'zero_crossing_rate'])
+    for key in feature_dict:
+      video_id = key
+      f_dict = feature_dict.get(key)
+      write_row(csv_writer, video_id, f_dict)
+
+
+def write_row(csv_writer, video_id, f_dict):
+  csv_writer.writerow([video_id] + [delete_outer_list(f_dict.get('chroma_stft').tolist())]
+                      + [delete_outer_list(f_dict.get('chroma_cqt').tolist())]
+                      + [delete_outer_list(f_dict.get('chroma_cens').tolist())]
+                      + [delete_outer_list(f_dict.get('melspectrogram').tolist())]
+                      + [delete_outer_list(f_dict.get('mfcc').tolist())]
+                      + [delete_outer_list(f_dict.get('rms').tolist())]
+                      + [delete_outer_list(f_dict.get('spectral_centroid').tolist())]
+                      + [delete_outer_list(f_dict.get('spectral_bandwidth').tolist())]
+                      + [delete_outer_list(f_dict.get('spectral_contrast').tolist())]
+                      + [delete_outer_list(f_dict.get('spectral_flatness').tolist())]
+                      + [delete_outer_list(f_dict.get('spectral_rolloff').tolist())]
+                      + [delete_outer_list(f_dict.get('poly_features').tolist())]
+                      + [delete_outer_list(f_dict.get('tonnetz').tolist())]
+                      + [delete_outer_list(f_dict.get('zero_crossing_rate').tolist())])
+
+
+def setup_output_csv():
+  with open("tfdataset.csv", 'w', newline='') as csvfile:
+    csv_writer = csv.writer(csvfile, delimiter=':', quoting=csv.QUOTE_MINIMAL)
+    csv_writer.writerow(
+      ['label', 'video_id', 'chroma_stft', 'chroma_cqt', 'chroma_cens', 'melspectrogram', 'mfcc', 'rms',
+       'spectral_centroid', 'spectral_bandwidth', 'spectral_contrast', 'spectral_flatness', 'spectral_rolloff',
+       'poly_features', 'tonnetz', 'zero_crossing_rate'])
+    return csv_writer
+
+
+def delete_outer_list(alist):
+  if len(alist) == 1 and isinstance(alist, list):
+    return delete_outer_list(alist[0])
+  return alist
+
+
+def main(csv_file, redo):
+  audio_list = parse_csv(file_structure + referDoc + csv_file)
+  download_from_list(file_structure + 'audio_balanced_train', audio_list, redo)
```

Comment:
Especially since this argument is re-used, consider extracting it out to a local variable in the main function. It'll help readability for the input parameter as well.

```
download_destination_dir = file_structure + 'audio_balanced_train'
download_from_list(download_destination_dir, audio_list, redo)
```